modules/vision/src/pose-estimation/vpPoseRGBD.cpp

```diff
@@ -0,0 +1,320 @@
+    plane_equation_estimated[3] = D;
+
+    // Compute error points to estimated plane
+    prev_error = error;
+    error = 0.0;
+    for (unsigned int i = 0; i < nPoints; i++) {
+      residues[i] = std::fabs(A * point_cloud_face[3 * i] + B * point_cloud_face[3 * i + 1] +
+                    C * point_cloud_face[3 * i + 2] + D) / sqrt(A * A + B * B + C * C);
+      error += residues[i] * residues[i];
+    }
+    error /= sqrt(error / total_w);
+  }
+
+  // Update final weights
+  tukey.MEstimator(vpRobust::TUKEY, residues, weights);
+
+  // Update final centroid
+  centroid.resize(3, false);
+  double total_w = 0.0;
+
+  for (unsigned int i = 0; i < nPoints; i++) {
+    centroid[0] += weights[i] * point_cloud_face[3 * i];
+    centroid[1] += weights[i] * point_cloud_face[3 * i + 1];
+    centroid[2] += weights[i] * point_cloud_face[3 * i + 2];
+    total_w += weights[i];
+  }
+
+  centroid[0] /= total_w;
+  centroid[1] /= total_w;
+  centroid[2] /= total_w;
+
+  // Compute final plane equation
+  double A = normal[0], B = normal[1], C = normal[2];
+  double D = -(A * centroid[0] + B * centroid[1] + C * centroid[2]);
+
+  // Update final plane equation
+  plane_equation_estimated[0] = A;
+  plane_equation_estimated[1] = B;
+  plane_equation_estimated[2] = C;
+  plane_equation_estimated[3] = D;
+}
+
+double computeZMethod1(const vpColVector& plane_equation, double x, double y) {
+  return -plane_equation[3] / (plane_equation[0]*x + plane_equation[1]*y + plane_equation[2]);
+}
+
+bool validPose(const vpHomogeneousMatrix& cMo) {
+  bool valid = true;
+
+  for (unsigned int i = 0; i < cMo.getRows() && valid; i++) {
+    for (unsigned int j = 0; j < cMo.getCols() && valid; j++) {
+      if (vpMath::isNaN(cMo[i][j])) {
+        valid = false;
+      }
+    }
+  }
+
+  return valid;
+}
+}
+
+/*!
+  Compute the pose of a planar object from corresponding 2D-3D point coordinates and depth map.
+  Depth map is here used to estimate the 3D plane of the object.
+
+  \param[in] depthMap : Depth map aligned to the color image from where \e corners are extracted.
+  \param[in] corners : Vector of 2D pixel coordinates of the object in an image.
+  \param[in] colorIntrinsics : Camera parameters used to convert \e corners from pixel to meters.
+  \param[in] point3d : Vector of 3D points corresponding to the model of the planar object.
+  \param[out] cMo : Computed pose.
+  \param[out] confidence : Pose estimation confidence index in range [0, 1]. When values are close to zero
+  it means that confidence is very low. Values close to 1 indicate a high level of confidence in the pose
+  estimation.
+
+  \return true if pose estimation succeed, false otherwise.
+ */
+bool vpPose::computePlanarObjectPoseFromRGBD(const vpImage<float> &depthMap, const std::vector<vpImagePoint> &corners,
+                                             const vpCameraParameters &colorIntrinsics, const std::vector<vpPoint> &point3d, vpHomogeneousMatrix &cMo, double &confidence)
```

Comment:
Are the estimated poses with this method reliable from your tests?

Generally, I don't like having to pass too much parameters. If `confidence` is necessary to evaluate if the estimated pose is valid, keep it otherwise I would suggest using something like `double *good_parameter_name=NULL`.

Reply:
It is very difficult to make appear the z-axis flipping phenomena using a real tag with the Realsense D435. I succeed by crumping the tag and observing a very small tag in the image. 

As mentioned in a previous discussion, this phenomena occurs also displaying this [image](https://user-images.githubusercontent.com/2388209/69619066-44b44080-103b-11ea-84f9-d5b41e7d6f1a.png) on my screen and detecting the upper right tag with the Realsense.

In both cases, results obtained with this method were reliable.

Ok passing a pointer.